datafusion/sql/src/expr/mod.rs

```diff
@@ -984,84 +984,141 @@ impl<S: ContextProvider> SqlToRel<'_, S> {
         Ok(Expr::Cast(Cast::new(Box::new(expr), dt)))
     }
 
-    fn sql_subscript_to_expr(
+    fn sql_compound_field_access_to_expr(
```

Comment:
The code to plan CompoundFieldAccess is directly copy/pasted from @goldmedal 's PR
- https://github.com/apache/datafusion/pull/13734